File: tests/fixtures/python/caller.py
"""Caller module that uses BaseClass and DerivedClass."""

from base import BaseClass, helper_function
from derived import DerivedClass


def process_objects() -> None:
    """Main processing function that calls methods from base and derived."""
    base = BaseClass("BaseObject")
    derived = DerivedClass("DerivedObject", 42)

    # Call methods on base class
    base_name = base.get_name()
    base_result = base.calculate(10, 20)
    base.print_info()

    # Call methods on derived class
    derived_name = derived.get_name()
    derived_result = derived.process_data(5, 10)
    id_value = derived.get_id()

    # Call standalone helper
    helper_result = helper_function(base_result)

    print(f"Processed: {base_name}, {derived_name}")


def calculate_total(x: int, y: int) -> int:
    """Calculate total using DerivedClass."""
    obj = DerivedClass("Calculator", 1)
    return obj.process_data(x, y)

File: tests/fixtures/typescript/caller.ts
/**
 * Caller module that uses BaseClass and DerivedClass.
 */

import { BaseClass, helperFunction } from './base';
import { DerivedClass } from './derived';

/**
 * Main processing function that calls methods from base and derived.
 */
export function processObjects(): void {
    const base = new BaseClass("BaseObject");
    const derived = new DerivedClass("DerivedObject", 42);

    // Call methods on base class
    const baseName = base.getName();
    const baseResult = base.calculate(10, 20);
    base.printInfo();

    // Call methods on derived class
    const derivedName = derived.getName();
    const derivedResult = derived.processData(5, 10);
    const idValue = derived.getId();

    // Call standalone helper
    const helperResult = helperFunction(baseResult);

    console.log(`Processed: ${baseName}, ${derivedName}`);
}

/**
 * Calculate total using DerivedClass.
 */
export function calculateTotal(x: number, y: number): number {
    const obj = new DerivedClass("Calculator", 1);
    return obj.processData(x, y);
}
